### PR TITLE
Measure flops optionally

### PR DESCRIPTION
**Pull request by shwetasalaria**

### CODE REVIEW

File: benchmarker/nvprof.py [hunk 1]
```diff
@@ -2,22 +2,22 @@
 import subprocess
 from benchmarker.util import abstractprocess
 
-exp1_re = re.compile('(\d+)\s+flop\_count\_sp\s+Floating\s+Point\s+Operations\(Single Precision\)\s+(?:(\d+)\s+){3}')
-exp2_re = re.compile('(\d+)\s+flop\_count\_sp\s+Floating\s+Point\s+Operations\(Single Precision\)\s+(?:(\d+\.\d+e[\+|\-]\d+)\s+){3}')
+nvprof_expr1_re = re.compile('(\d+)\s+flop\_count\_sp\s+Floating\s+Point\s+Operations\(Single Precision\)\s+(?:(\d+)\s+){3}')
```
Comment (vatai) [inline benchmarker/nvprof.py:5]: What does this (and the next) `re` do? What expression/value do they match? Flops? Put that in the name of the regexp!

File: benchmarker/benchmarker.py [hunk 2]
```diff
@@ -33,7 +33,8 @@ def parse_basic_args(argv):
     parser.add_argument("--problem_size", default=None)
     parser.add_argument("--batch_size", default=None)
     parser.add_argument("--power_sampling_ms", type=int, default=100)
-    # parser.add_argument('--misc')
+    parser.add_argument("--flops", action="store_true")
+    #parser.add_argument('--misc')
```
Comment (vatai) [inline benchmarker/benchmarker.py:37]:
```suggestion
    # parser.add_argument('--misc')
```
The right way to comment is to put a space after the comment! :P :P :P 

More serious questions: why are there two `--flop` args parts? Do both the "outer" and the "inner" benchmarker need a `--flop` flag? What does the `--flop` flag do in the inner/core benchmarker?

Other question: Do the tests pass? And did you test it with both with and without the `--flop` flag? Do they work in both cases!? :)

**Review by vatai — COMMENTED** [review]
Generally good/straight forward. Have a few questions, but yeah, this is totally mergable!

File: benchmarker/nvprof.py (hunk 1 above)
Comment (shwetasalaria) [reply benchmarker/nvprof.py:5]: The first regex captures operations in whole number and the second one captures operations written in scientific notation. Ideally, I wanted to use one regex that captures both but it was not working so kept as it for the time being but will update once I figure that out.
Ok, making regexp names highlighting what I mentioned above.

Comment (vatai) [reply benchmarker/nvprof.py:5]: yey! :partying_face:

File: benchmarker/benchmarker.py (hunk 2 above)
Comment (shwetasalaria) [reply benchmarker/benchmarker.py:37]: Initially, I did not add it in the "inner" benchmarker but it raised an error:
benchmarker.py: error: unrecognized arguments: --flops

I tested it both with and without the --flop flag, works fine. The tests also pass OK.

Comment (vatai) [reply benchmarker/benchmarker.py:37]: Ok. I kinda get it: the inner `--flops` doesn't do anything... it's just there so you don't get an error, right? But then how do the other non-inner benchmarking parameters not causing the error? Are the same way?

Comment (vatai) [reply benchmarker/benchmarker.py:37]: Ok. Checked it out. It seems like they are handled in a similar way (just added and not used)... But you see what's the problem with that, right? We got to maintain 2 lists of "outer parameters" which sounds like something that could be solved somehow in a more elegant way. So you should add a `TODO(Shwerta):` there... maybe add an issue :)

**Review by vatai — CHANGES_REQUESTED** [review]

**Review by vatai — APPROVED** [review]

**Review by vatai — APPROVED** [review]

**Pull request merged**